README.md

```diff
@@ -76,6 +89,124 @@ const App = () => (
 )
 ```
 
+## Props
+
+This is the list of props that you need to pass to the component.
+
+### getIcon
+
+> `function(state: object)` | optional, default icon is provided based on state object
+
+This function decides what icon to show based on the current state of the component.
+
+### getMessage
+
+> `function(icon: string, state: object)` | optional, default message is provided based on the icon and state object.
+
+This function decided what message to show based on the icon (returned from getIcon prop) and the current state of the component.
+
+### getUrl
+
+> `function({})` | optional, no useful default
+
+This function is called as soon as the component enters the viewport and is provided with an element from `props.srcSet`.
```

Comment:
getUrl used to generate urls based on `width` and `format` if `srcSet` doesn't provide `src` field